### PR TITLE
Fix commit modal overflow scroll

### PR DESCRIPTION
**Pull request by dweberdev**
**Issue:** Not able to vertically scroll commit modal on small screens

### CODE REVIEW

**Comment by dospore** [comment]
Was this a fix for mobile? not that its likely wed get a screen like this but might as well fix for all 
[

](url)

**Comment by dweberdev** [comment]
> Was this a fix for mobile? not that its likely wed get a screen like this but might as well fix for all

Yes, users were not able to scroll or close the modal on screen size under 700px

**Comment by dweberdev** [comment]
> Was this a fix for mobile? not that its likely wed get a screen like this but might as well fix for all Screen Shot 2022-04-13 at 8 00 29 am

Good catch, updated PR

File: components/General/TWModal/index.tsx [hunk 1]
```diff
@@ -35,6 +35,11 @@ export const TWModal: React.FC<TWModalProps> = ({ open, onClose, size = 'default
                         <Dialog.Overlay className="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity" />
                     </Transition.Child>
 
+                    {/* This element is to trick the browser into centering the modal contents. */}
+                    <span className="inline-block h-screen align-middle" aria-hidden="true">
+                        &#8203;
+                    </span>
+
```
Comment (dospore) [inline components/General/TWModal/index.tsx:42]: whould be interested to know how this works if you could explain

**Review by dospore — APPROVED** [review]
lgtm besides the confused comment

File: components/General/TWModal/index.tsx (hunk 1 above)
Comment (dweberdev) [reply components/General/TWModal/index.tsx:42]: This is the recommended implementation from headless ui docs. We had this code implemented previously but I didn't fully understand what it was doing and the idea of tricking the browser kept bugging me so I replaced with traditional css modal positioning while working on the commit modal.

My implementation led to some overlay positioning bugs on screens smaller than the modal height.

I did spend time trying to solve the initial issue just with css... which led me to stackoverflow... then github issues.. finally back to their docs and realize it was their recommendation.

**Comment by dweberdev** [comment]
> lgtm besides the confused comment. edit: changed my mind, can we show the scroll bar on the modal as well?

You mean add a set height to the modal and scroll the content inside the modal?

If we go this route I would need to add media query to the height of the modal to avoid having 2 scroll bars, one for the content and one for the modal.

**Comment by dospore** [comment]
> > lgtm besides the confused comment. edit: changed my mind, can we show the scroll bar on the modal as well?
> 
> You mean add a set height to the modal and scroll the content inside the modal?
> 
> If we go this route I would need to add media query to the height of the modal to avoid having 2 scroll bars, one for the content and one for the modal.

hmmmm, ok this can do for now, might get updated if we move away from headless UI and tailwind

**Review by dospore — APPROVED** [review]

**Pull request merged**